Dockerfile

```diff
@@ -1,10 +1,22 @@
-FROM python:3.10-alpine
+FROM python:3.10-alpine as builder
 
 WORKDIR /app
 
 COPY . .
 
-RUN pip install -r requirements.txt
+RUN pip install --user -r requirements.txt
+
+FROM python:3.10-alpine
+
+RUN adduser -D appuser
+USER appuser
+
+WORKDIR /app
+
+COPY --from=builder /root/.local /home/appuser/.local
+COPY --from=builder /app /app
```

Comment:
On both COPY don't forget to --chown=adduser:adduser to set permissions on the file. Verify the perms by executing a bash terminal in a local container instance of this image if you want or use the dive CLI